packages/headless/src/app/engine.ts

```diff
@@ -99,6 +100,23 @@ export interface CoreEngine<
   disableAnalytics(): void;
 }
 
+export type CoreEngineNext<
+  State extends object = {},
+  ExtraArguments extends ThunkExtraArguments = ThunkExtraArguments,
+> = Omit<CoreEngine<State, ExtraArguments>, 'state' | 'store'> & {
+  /**
+   * The readonly internal state of the headless engine.
+   *
+   * @internal
+   */
+  readonly [stateKey]: State & CoreState;
+
+  /**
+   * The readonly global headless engine configuration
+   */
+  readonly configuration: EngineConfiguration;
```

Comment:
I'm exposing this here because there's not configuration controller.

Also in Atomic, we retrieve information from the configuration state in [query-error-common.tsx](https://github.com/coveo/ui-kit/pull/3890/files#diff-9d7602efb11be67faba116b145f0fc95a945d94657859c4e7d971637672c7a3aL74)

Reply:
Should we have a configuration controller?

Reply:
I don't think so, no; if you need to adjust the engine configuration, you would normally re-initialize the engine.